clients/drcachesim/scheduler/scheduler_impl.h

```diff
@@ -261,6 +262,20 @@ template <typename RecordType, typename ReaderType> class scheduler_impl_tmpl_t
         uint64_t last_run_time = 0;
     };
 
+    // XXX i#6831: Should this live entirely inside the dynamic subclass?
+    // We would need to make part of ini() virtual with subclass overrides.
+    struct workload_info_t {
+        explicit workload_info_t(int output_limit)
+            : output_limit(output_limit)
+        {
+            live_outputs = std::unique_ptr<std::atomic<int>>(new std::atomic<int>());
```

Comment:
Why does this need to be a unique_ptr to an std::atomic and not just an std::atomic?

Reply:
Added the same comment by the other unique_ptr to std::atomics in this file:

        // We use a unique_ptr to make this moveable for vector storage.